cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -2789,20 +2789,19 @@ static char mappings[] = {'0', '1', '2', '3', '0', '1', '2', '0', '0',
 // the input string followed by a phonetic code. Characters that are not alphabetic are
 // ignored. If expression evaluates to the null value, null is returned.
 //
-// The soundex algorith works with the following steps:
+// The soundex algorithm works with the following steps:
 //    1. Retain the first letter of the string and drop all other occurrences of a, e, i,
-//    o, u, y, h, w.
+//    o, u, y, h, w. (let's call them special letters)
 //    2. Replace consonants with digits as follows (after the first letter):
 //        b, f, p, v → 1
 //        c, g, j, k, q, s, x, z → 2
 //        d, t → 3
 //        l → 4
 //        m, n → 5
 //        r → 6
-//    3. If two or more letters with the same number are adjacent in the original name
-//    (before step 1), only retain the first letter; also two letters with the same number
-//    separated by 'h' or 'w' are coded as a single number, whereas such letters separated
-//    by a vowel are coded twice. This rule also applies to the first letter.
+//    3. If two or more letters with the same number were adjacent in the original name
+//    (before step 1), or adjacent for any intervening from any special letters, then omit
```

Comment:
What does "or adjacent for any intervening from any special letters" mean?

Reply:
simply adjacent:  **BB** or **CC** or **DD**
adjacent with special intervening: **B**A**B**, **C**H**C**, **D**AAAAA**D**

For any of these cases, the result is always B, C, D

Reply:
Do we have a test case for DAAAAD returning D000? I dont see the code behaving that way

Reply:
You right, i made a mistake, I changed the documentation thinking in intervening with H and W from MySQL. 
Oracle remove only if is same number adjacent in original text, i removed this from documentation